TwoFactorAuth/Command/TfaDisable.php

```diff
@@ -48,23 +49,18 @@ public function __construct(
     protected function configure()
     {
         $this->setName('security:tfa:disable');
-        $this->setDescription('Globally disable two factor auth');
+        $this->setDescription('[DEPRECATED] Globally disable two factor auth');
 
         parent::configure();
     }
 
     /**
+     * @inheritDoc
+     *
      * @SuppressWarnings("PHPMD.UnusedFormalParameter")
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $this->config->saveConfig(
-            TfaInterface::XML_PATH_ENABLED,
-            '0',
-            'default',
-            0
-        );
-
-        $this->cacheManager->flush(['config']);
+        trigger_error('Disabling 2FA is deprecated', E_USER_DEPRECATED);
```

Comment:
This is our standard way of removing functionality as far as I know

Reply:
Even for the monolith this isn't how it would work. The monolith would be flagged via annotation and the behavior and contracts would still work until it is actually removed. 

In this case the whole module is basically being rewritten and will be broken both functionally and by code contract. Plus, we are going to be releasing this with a new major version so we can break and remove essentially whatever we want.

Reply:
adding E_USER_DEPRECATED is the standard way to deprecate stuff for core. The classes that were updated with the error still work